### ISSUE TITLE
Cannot complete microservice challenge

### ISSUE THREAD

**Issue by idietmoran** [freeCodeCamp/freeCodeCamp#17078]
NOTE: Answers below

#### Challenge Name
https://www.freecodecamp.org/challenges/file-metadata-microservice

#### Issue Description
Submitting links to the challenge will not close it out 
http://file-metadata-fcc-.herokuapp.com/
https://github.com/idietmoran/file-metadata

#### Browser Information

* Browser Name, Version: Firefox Developer Edition 60.0b14
* Operating System: Windows 10
* Mobile, Desktop, or Tablet: Desktop

**Comment by Asjas** [comment]
Hi @idietmoran, 

Please note the challenge states you need to deploy to glitch and not heroku.

**Comment by idietmoran** [comment]
thanks, wasn't aware that glitch was required.

and the heroku app is working for me which is weird.

**Comment by idietmoran** [comment]
@Asjas nope, just submitted with glitch as well.

http://fcc-file-metadata-.glitch.me/

still nothing. I'm not sure if it's bugged or not

**Comment by scissorsneedfoodtoo** [comment]
@idietmoran, I just tried it on the freeCodeCamp production site and the beta site and it seems to work. But I noticed a mistake with the URL from your last comment. The ones that worked for me were:

```
http://fcc-file-metadata.glitch.me/
https://github.com/idietmoran/file-metadata
```

Could you try submitting those and see if they work for you, too?

**Comment by Asjas** [comment]
@idietmoran Yes, it seems you are typing an extra hyphen in the herokuapp url. I didn't notice it.

Try what @scissorsneedfoodtoo suggests.

**Comment by idietmoran** [comment]
@scissorsneedfoodtoo @Asjas those aren't my projects, my projects are at the ones with the hypen at the end.

**Comment by scissorsneedfoodtoo** [comment]
@idietmoran, I see. So `http://file-metadata-fcc-.herokuapp.com/` with the hyphen is the correct URL? When I visit the page I get a ERR_NAME_NOT_RESOLVED error. Do you get the same error when you visit the page? Or does it load correctly?

**Comment by Asjas** [comment]
@idietmoran I don't think that is a correct naming convention. Because your heroku app is still offline.

@scissorsneedfoodtoo It's offline for me as well.

**Comment by Asjas** [comment]
@idietmoran Rather create a new name on heroku instead of using the fcc name with just a hyphen at the end. Just create a heroku instance with any name.

But the main thing is, this isn't a problem with the codebase itself. @scissorsneedfoodtoo tested the project submission and it worked. Issues on Github are reserved for problems with the code or the actual platform. Not for normal support.

@idietmoran I would suggest you ask for assistance on the [freeCodeCamp gitter room](https://gitter.im/FreeCodeCamp/HelpBackEnd) as there is nothing to fix on the actual code.

**Comment by scissorsneedfoodtoo** [comment]
That's interesting. I'm not sure why your page isn't loading for us, then. I also rechecked with Chrome and tried with Firefox on my end and I'm still getting the same error. But when I tried it on my phone with Chrome your website loaded, so I don't know what's going on here.

All that said, I think the problem is that, like @Asjas said, the URL for your project doesn't follow the normal naming convention. If you check this [URL validator](http://formvalidation.io/validators/uri/) `http://a.b-.co` is considered invalid. However, `http://a.b.co` is fine. Similarly, `http://file-metadata-fcc-.herokuapp.com/` is invalid while `http://file-metadata-fcc.herokuapp.com/` passes.

I'm also going to suggest changing the name of your project. I know it can be a pain thinking of a name and updating something on Heroku, but it does seem like enforcing conventional URL naming practices is for the best.

If you do that and still can't complete this challenge, then let's revisit this issue and see what the problem might be.
